android/cli/commands/_build.js

```diff
@@ -2794,110 +2802,67 @@ AndroidBuilder.prototype.copyResources = function copyResources(next) {
-				this.unmarkBuildDirFile(assetCryptDest);
-				fs.writeFileSync(
-					assetCryptDest,
-					ejs.render(fs.readFileSync(path.join(this.templatesDir, 'AssetCryptImpl.java')).toString(), {
-						appid: this.appid,
-						encryptedAssets: out
+		// NOTE: maintain 'build.android.titaniumprep' hook for remote encryption policy.
+		this.cli.createHook('build.android.titaniumprep', this, async next => {
+			try {
+				await Promise.all(
+					jsFilesToEncrypt.map(async file => {
+						const from = path.join(this.buildAssetsDir, file);
+						const to = path.join(this.buildBinAssetsResourcesDir, file);
+
+						this.logger.debug(__('Encrypting: %s', from.cyan));
+						await fs.ensureDir(path.dirname(to));
+						this.unmarkBuildDirFile(to);
+						return await cloak.encryptFile(from, to);
 					})
 				);
 
-				done();
-			}.bind(this));
-		});
-		let args = [ this.tiapp.guid, this.appid, this.buildAssetsDir ].concat(jsFilesToEncrypt);
-		if (process.platform === 'win32') {
-			const fileListing = path.join(this.buildDir, 'titanium_prep_listing.txt');
-			args = [ this.tiapp.guid, this.appid, this.buildAssetsDir, '--file-listing', fileListing ];
-			fs.writeFileSync(fileListing, jsFilesToEncrypt.join('\n'));
-		}
-
-		const opts = {
-				env: appc.util.mix({}, process.env, {
-					// we force the JAVA_HOME so that titaniumprep doesn't complain
-					JAVA_HOME: this.jdkInfo.home
-				})
-			},
-			fatal = function fatal(err) {
-				this.logger.error(__('Failed to encrypt JavaScript files'));
-				err.msg.split('\n').forEach(this.logger.error);
-				this.logger.log();
-				process.exit(1);
-			}.bind(this);
-
-		titaniumPrepHook(
-			path.join(this.platformPath, titaniumPrep),
-			args.slice(0),
-			opts,
-			function (err) {
-				if (!err) {
-					return next();
-				}
+				this.logger.info('Writing encryption key...');
+				await cloak.setKey('android', this.abis, path.join(this.buildDir, 'libs'));
 
-				if (process.platform !== 'win32' || !/jvm\.dll/i.test(err.msg)) {
-					fatal(err);
-				}
-
-				this.logger.debug(__('32-bit titanium prep failed, trying again using 64-bit'));
-				titaniumPrep = 'titanium_prep.win64.exe';
-				titaniumPrepHook(
-					path.join(this.platformPath, titaniumPrep),
-					args,
-					opts,
-					function (err) {
-						if (err) {
-							fatal(err);
+				// Generate 'AssetCryptImpl.java' from template.
+				const assetCryptDest = path.join(this.buildGenAppIdDir, 'AssetCryptImpl.java');
+				this.unmarkBuildDirFile(assetCryptDest);
+				await fs.writeFile(
+					assetCryptDest,
+					ejs.render(
+						await fs.readFile(path.join(this.templatesDir, 'AssetCryptImpl.java'), 'utf8'),
+						{
+							appid: this.appid,
+							assets: jsFilesToEncrypt,
+							salt: cloak.salt
 						}
-						next();
-					}
+					)
 				);
-			}.bind(this)
-		);
+
+				next();
+			} catch (e) {
+				next(new Error('Could not encrypt assets!\n' + e));
+			}
+		})(_ => next());
```

Comment:
<!--
  0 failure: 
  1 warning:  android/cli/comma...
  
  
  DangerID: danger-id-Danger;
  File: android/cli/commands/_build.js;
  Line: 2865;
-->

- :warning: android/cli/commands/_build.js line 2865 – '_' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)